### PR TITLE
If container exits with 125 podman should exit with 125

### PR DESCRIPTION
**Pull request by rhatdan**
fixes: https://github.com/containers/podman/issues/11540

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@jwhonce Any remembrance why you added this code?
```
diff --git a/cmd/podman/root.go b/cmd/podman/root.go
index 371ded9a8..cf712aa90 100644
--- a/cmd/podman/root.go
+++ b/cmd/podman/root.go
@@ -90,13 +90,6 @@ func init() {
 func Execute() {
        if err := rootCmd.ExecuteContext(registry.GetContextWithOptions()); err != nil {
                fmt.Fprintln(os.Stderr, formatError(err))
-       } else if registry.GetExitCode() == registry.ExecErrorCodeGeneric {
-               // The exitCode modified from registry.ExecErrorCodeGeneric,
-               // indicates an application
-               // running inside of a container failed, as opposed to the
-               // podman command failed.  Must exit with that exit code
-               // otherwise command exited correctly.
-               registry.SetExitCode(0)
        }
        os.Exit(registry.GetExitCode())
 }
```

File: cmd/podman/registry/registry.go [hunk 1]
```diff
@@ -23,12 +24,10 @@ type CliCommand struct {
 	Parent  *cobra.Command
 }
 
-const ExecErrorCodeGeneric = 125
-
 var (
 	cliCtx          context.Context
 	containerEngine entities.ContainerEngine
-	exitCode        = ExecErrorCodeGeneric
+	exitCode        = define.ExecErrorCodeGeneric
```
Comment (Luap99) [inline cmd/podman/registry/registry.go:30]: the default exit code should be 0

File: cmd/podman/root.go [hunk 2]
```diff
@@ -90,13 +90,6 @@ func init() {
 func Execute() {
 	if err := rootCmd.ExecuteContext(registry.GetContextWithOptions()); err != nil {
 		fmt.Fprintln(os.Stderr, formatError(err))
```
Comment (Luap99) [inline cmd/podman/root.go:92]: when there is an error check `registry.GetExitCode() == 0` and set it to registry.ExecErrorCodeGeneric in this case

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM but tests are unhappy and one comment that needs to be addressed.

**Comment by rhatdan** [comment]
@Luap99 PTAL

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**